src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -164,6 +179,69 @@ private float getSentimentScore(String feedback) throws IOException {
     return queryList;
   }
 
+  private double getToxicityScore(String comment) throws IOException {
+    HttpURLConnection commentAnalyzerConnection = creatingPostRequestCommentAnalyzer();
+    // Create JSON request.
+    JSONObject jsonObjectCommentAnalyzer = jsonObjectCommentAnalyzerRequest(comment);
+
+    OutputStreamWriter writer = new OutputStreamWriter(commentAnalyzerConnection.getOutputStream());
+    writer.write(jsonObjectCommentAnalyzer.toString());
+    writer.close();
+
+    int responseCode =
+        commentAnalyzerConnection.getResponseCode(); // New items get NOT_FOUND on POST.
+    if (responseCode == HttpURLConnection.HTTP_OK
+        || responseCode == HttpURLConnection.HTTP_NOT_FOUND) {
+      StringBuilder response = new StringBuilder();
+      String line;
+
+      // Read input data stream.
+      BufferedReader reader =
+          new BufferedReader(new InputStreamReader(commentAnalyzerConnection.getInputStream()));
+      while ((line = reader.readLine()) != null) {
+        response.append(line);
+      }
+      reader.close();
+      JSONObject responseJsonObject = new JSONObject(response.toString());
+      double toxicityValue =
+          responseJsonObject
+              .getJSONObject("attributeScores")
+              .getJSONObject("TOXICITY")
+              .getJSONObject("summaryScore")
+              .getFloat("value");
+      return toxicityValue;
+    } else {
+      throw new RuntimeException("Cannot get toxicity score.");
+    }
+  }
+
+  private HttpURLConnection creatingPostRequestCommentAnalyzer() throws IOException {
+    String apiKey = "AIzaSyBnjF0OVUD3BGiuYFMSVe1_g134AKz3xQY";
```

Comment:
would this not be a risk to put this on the public github?

Reply:
I am not sure to be honest, I will check with Travis and Luke on this one.